comms/dht/src/rpc/service.rs

```diff
@@ -68,9 +68,27 @@ impl DhtRpcServiceImpl {
             let iter = peers
                 .into_iter()
                 .filter_map(|peer| {
-                    let peer_info =
+                    let mut peer_info =
                         UnvalidatedPeerInfo::from_peer_limited_claims(peer, max_claims, max_addresses_per_claim);
 
+                    // Filter out all identity claims with invalid signatures
+                    let count = peer_info.claims.len();
```

Comment:
How does this happen? Peers that you are streaming are, by definition, valid. 

If this is happening, then we should be removing invalidated addresses from the peer db.
If this is an issue, happy to keep this here but we should add a TODO comment saying that we should remove invalidated peer signatures.

Reply:
This is more _defence-in-depth_ than anything else, closing a potential attack vector. I will remove it, but we should actually ban the peer for offences in `UnvalidatedPeerInfo::from_peer_limited_claims` as well as providing invalid signatures. What is your opinion?